apps/vessel-history/features/map/map.hooks.ts

```diff
@@ -53,3 +64,93 @@ export interface Viewport extends LatLon {
   longitude: number
   zoom: number
 }
+
+export const openFishingMap = (datasets, allGFWIds, viewport) => {
+  // colors used in the map
+  const colors = ['#f4511f', '#33b679', '#f09300', '#ffea00', '#9ca4ff']
+  const presenceDataviews = [
+    {
+      id: 'fishing-ais',
+      config: {
+        visible: false,
+      },
+    },
+    {
+      id: 'vms',
+      config: {
+        visible: false,
+      },
+    },
+    {
+      id: 'basemap-labels',
+      config: {
+        visible: true,
+        locale: 'en',
+      },
+    },
+
+    {
+      id: 'context-layer-rfmo',
+      config: {
+        visible: true,
+      },
+    },
+    {
+      id: 'context-layer-mpa',
+      config: {
+        visible: true,
+      },
+    },
+    {
+      id: 'context-layer-eez',
+      config: {
+        visible: true,
+      },
+    },
+  ]
+
+  const instances: UrlDataviewInstance<GeneratorType>[] = allGFWIds.map((GFWId, index) => {
+    const vesselDataset = decodeURIComponent(GFWId[0])
+    const infoDataset = datasets.find((dataset) => dataset.id === vesselDataset)
+    const trackDataset = getRelatedDatasetByType(infoDataset, DatasetTypes.Tracks)
+    const eventsRelatedDatasets = getRelatedDatasetsByType(infoDataset, DatasetTypes.Events)
+
+    const eventsDatasetsId =
+      eventsRelatedDatasets && eventsRelatedDatasets?.length
+        ? eventsRelatedDatasets.map((d) => d.id)
+        : []
+
+    if (infoDataset || trackDataset) {
+      const vesselDataviewInstance = {
+        ...getVesselDataviewInstance(
+          { id: GFWId[1] },
+          {
+            trackDatasetId: trackDataset?.id,
+            infoDatasetId: infoDataset?.id,
+
+            ...(eventsDatasetsId.length > 0 && { eventsDatasetsId }),
+          },
+          TEMPLATE_VESSEL_DATAVIEW_ID
+        ),
+
+        config: {
+          color: colors[index],
+        },
+      }
+      return vesselDataviewInstance
+    }
+    return null
+  })
+
+  const urlJson: BaseUrlWorkspace = {
+    latitude: viewport.latitude,
+    longitude: viewport.longitude,
+    zoom: viewport.zoom,
+    start: DEFAULT_WORKSPACE.availableStart,
+    end: DEFAULT_WORKSPACE.availableEnd,
```

Comment:
Why not use the current user date filter selection and leave the DEFAULT_WORKSPACE as a fallback if the user didn't update the filter?

Reply:
I dont use the current because:
1 is not defined yet
2  will use the voyage dates and is a period so small